### PR TITLE
check pid before aborting start

### PR DESCRIPTION
**Pull request by nerdrew**

### CODE REVIEW

File: lib/debug_socket.rb [hunk 1]
```diff
@@ -16,7 +17,10 @@ def self.logger
   end
 
   def self.start(path)
-    raise "debug socket thread already running" if @thread
+    pid = Process.pid
+    raise "debug socket thread already running for this process" if @thread && @pid == pid
```
Comment (embark) [inline lib/debug_socket.rb]: doesn't this technically allow a fork to open infinite debug sockets?

Comment (embark) [inline lib/debug_socket.rb]: Also, don't the `@thread`s clobber each other?

File: spec/debug_socket_spec.rb [hunk 2]
```diff
@@ -40,9 +40,33 @@
...
+          another_socket = UNIXSocket.new(another_path)
+          another_socket.write("Thread.list.each(&:wakeup)")
+          another_socket.close_write
+          expect(another_socket.read).to match(/Thread/)
+          Process.wait(child, Process::WNOHANG)
+        else
+          DebugSocket.start(another_path)
+          sleep
+          sleep(1)
+          DebugSocket.stop
```
Comment (embark) [inline spec/debug_socket_spec.rb]: Maybe add to the test that the debug socket thread from `start` was actually stopped?

File: lib/debug_socket.rb (hunk 1 above)
Comment (nerdrew) [inline lib/debug_socket.rb]: When MRI forks, it kills all other threads, so if process-A starts a debug socket, then forks, the new process-B child will not have a running `@thread`.

Comment (nerdrew) [inline lib/debug_socket.rb]: Yes, if you do infinite forks, you could open infinite debug sockets. At least that is what I was trying to do. I might have messed up the logic though.

Comment (embark) [inline lib/debug_socket.rb]: I guess I mean, the same fork can open infinite debug sockets. Because `@pid == pid` will always be false.

Comment (embark) [inline lib/debug_socket.rb]: WAITAMINUTE `@pid = pid` is right below this, I completely missed that. changes LGTM :P

**Pull request merged**